astropy/units/quantity.py

```diff
@@ -275,6 +275,14 @@ def __new__(cls, value, unit=None, dtype=None, copy=True, order=None,
     def __array_finalize__(self, obj):
         self._unit = getattr(obj, '_unit', None)
 
+        # Copy info (except for a scalar output for performance reasons) if the
+        # original had `info` defined.  Because of the way the InfoDescriptor
+        # works, `'info' in obj.__dict__` is False until the `info` attribute
+        # is accessed or set.  Note that `obj` can be an ndarray which doesn't
+        # have a `__dict__`.
+        if self.ndim > 0 and hasattr(obj, '__dict__') and 'info' in obj.__dict__:
```

Comment:
I experimented a little, and this seems to be the fastest:
```
if self.ndim > 0 and 'info' in getattr(q, '__dict__', ()):
```
is a little faster still, especially for the case of quantities itself (which I think is most important.

But, actually, is there sense in copying `info` if the object did not even have a `unit`? So, maybe just:
```
try:
    self._unit = obj._unit
except AttributeError:
    pass
else:
    if self.ndim > 0 and 'info' in obj.__dict__:
        self.info = obj.info
```